client/components/MusicWidget.jsx

```diff
@@ -1,18 +1,32 @@
-import React, { useRef, useState } from 'react';
+import React, { useRef, useState, useEffect } from 'react';
 import YoutubeFunctionality from './YoutubePlay';
 import { makeWidgetMovable } from '../functions/widgetMove';
 
 // Music Widget - holds the form and takes in YoutubePlay component
 export default function MusicWidget() {
   const widgetRef = useRef(null);
   const [videoId, setVideoId] = useState('');
+  const widgetId = 'MusicWidget';
 
-  const handleWidgetMovement = makeWidgetMovable(widgetRef);
+  const handleWidgetMovement = makeWidgetMovable(widgetRef, widgetId);
 
   function handleVideoChange(newVideo) {
     setVideoId(newVideo);
   }
 
+  function loadWidgetPosition() {
+    const coords = JSON.parse(localStorage.getItem(`widgetCoords-${widgetId}`));
```

Comment:
I suggest you pull all this localStorage code into a separate module. Then you won't have to worry that you got it right everywhere.

The module might have functions like `readWidgetCoords(widgetId)`, etc.

Reply:
will modularize it once I know it's all working, thank you king 👑